src/graph.erl

```diff
@@ -0,0 +1,270 @@
+-include("records.hrl").
+
+%%%% @todo You can use/rename any of those, your business
+
+%%generate_id() ->
+%%    Id = << <<Y>> ||<<X:4>> <= crypto:hash(md5, term_to_binary(make_ref())), Y <- integer_to_list(X,16)>>,
+%%    case persistence:if_exists(Id) of
+%%        {ok, 0} -> Id;
+%%        {ok, 1} -> generate_id();
+%%        {error, Reason} -> {error, Reason}
+%%    end.
+%%
+%%get_vertex(Id) ->
+%%    persistence:get(Id).
+%%
+%%get_vertices()->
+%%    persistence:keys("*").
+%%
+%%add_user(Name) ->
+%%    Id = generate_id(),
+%%    Json = json_utils:encode(#{<<"type">> => <<"user">>, <<"id">> => Id, <<"name">> => Name}),
+%%    Response = persistence:set(Id, Json),
+%%    case Response of
+%%        {error, Reason} -> {error, Reason};
+%%        {ok, _Result} -> {ok, Id}
+%%    end.
+%%
+%%update_user(Id, NewName) ->
+%%    Json = json_utils:encode(#{<<"type">> => <<"user">>, <<"id">> => Id, <<"Name">> => NewName}),
+%%    Response = persistence:set(Id, Json),
+%%    case Response of
+%%        {error, Reason} -> {error, Reason};
+%%        {ok, _Result} -> ok
+%%    end.
+%%
+%%delete(Id) ->
+%%    Response = persistence:del(Id),
+%%    case Response of
+%%        {error, Reason} -> {error, Reason};
+%%        {ok, _Result} -> ok
+%%    end.
+%%
+%%children_key(Key) when is_binary(Key) == true ->
+%%    <<Key/binary, "/children">>;
+%%children_key(Key) -> children_key(binary:list_to_bin(Key)).
+%%
+%%add_user_to_group(User, Group) ->
+%%    persistence:set_add(children_key(Group), User).
+%%
+%%remove_user_from_group(User, Group) ->
+%%    persistence:set_remove(children_key(Group), User).
+%%
+%%is_member_of_group(User, Group) ->
+%%    {ok, Result} = persistence:set_is_member(children_key(Group), User),
+%%    Result.
+%%
+%%list_group_users(Group) ->
+%%    {ok, Result} = persistence:set_list_members(children_key(Group)),
+%%    Result.
+%%
+%%
+%%add_group(Name) ->
+%%    Id = generate_id(),
+%%    Json = json_utils:encode(#{<<"type">> => <<"group">>, <<"id">> => Id, <<"name">> => Name}),
+%%    Response = persistence:set(Id, Json),
+%%    case Response of
+%%        {error, Reason} -> {error, Reason};
+%%        {ok, _Result} -> {ok, Id}
+%%    end.
+%%
+%%
+%%
+%%create_edge(Parent, Child, Permissions) ->
+%%    Flag = case {vertex_exists(Parent), vertex_exists(Child)} of
+%%               {false, _} -> false;
+%%               {_, false} -> false;
+%%               _ -> true
+%%           end,
+%%    case Flag of
+%%        false -> {error, "At least one of nodes doesn't exist"};
+%%        true ->
+%%            persistence:add_parent(Child, Parent, Permissions),
+%%            persistence:add_child(Parent, Child, Permissions),
+%%            ok
+%%    end.
+%%
+%%vertex_exists(Key) ->
+%%    case persistence:if_exists(Key) of
+%%        {ok, 0} -> false;
+%%        {ok, 1} -> true;
+%%        _ -> error
+%%    end.
+
+
+%%%% Parent, Child, Vertex to wszystko ID-ki, jesli wolisz mozesz zmienic nazwy na ParentId itd., jak uwazasz
```

Comment:
Dobrze by było jednak usunąć ten zakomentowany blok kodu. Zawsze można do niego wrócić patrząc na wcześniejsze commity, skoro nie usuwamy branchy

Reply:
W zasadzie zostawiłem to tobie. I tak masz implementować funkcje do krawędzi, więc będziesz się grzebał w tym module, więc rób z tym co chcesz